### ISSUE TITLE
Expose package types for external use.

### ISSUE THREAD

**Issue by anthonyshew** [stytchauth/stytch-node#118]
Hey, Stytch.

Trying to get set up with you guys and wanted to use the internal types so I could get stronger types across my apps. However, I'm finding that the types aren't exposed as I would have hoped for.

My best case scenario would have been to be able to re-export all the internal types using `stytch/types`. I also tried `stytch/types/lib/index` to no avail.
Another option I would have been happy with would have been to re-export the entire package (`export * from 'stytch`) if it included the types.

It does look like, though, that the `dist` doesn't include the types so perhaps this behavior is to be expected: 
![image](https://user-images.githubusercontent.com/35677084/163664819-369b75b3-8bb0-40b4-bc0e-d59f5e51e497.png)

For now, I've accomplished reusable types around my project by creating them myself:
```
export const authenticateMagicLink = (token: string) => {
	try {
		return client.magicLinks.authenticate(token, {
			session_duration_minutes: 43800 // A month
		})
	} catch (error) {
		handleError(error)
	}
}

export type AuthenticateMagicLinkReturnType = Awaited<ReturnType<typeof authenticateMagicLink>>
```

But it would still be nice to have them ready-to-go without having to draw them up myself, I suppose.

Thanks for your great work and hope to see these types be available! (or the reason why they aren't 😊)

EDIT: And as I continue to work on this, I keep running into more spots where I'd love to use the package's types. Authentication responses, OAuth factor, requests and responses, etc.

**Comment by chris-stytch** [comment]
Heya @anthonyshew ! 

Thanks for providing this feedback, super helpful. I'm going to open up a ticket to take a look at this for the next major release.

Are there any other areas of `stytch-node` that you'd like some better tooling around?

**Comment by Max-Stytch** [comment]
Hey @anthonyshew - following up on this. You should be able to import the types directly from the individual `d.ts` files - e.g.
```typescript
import {
  AuthenticateResponse as MagicLinksAuthenticateResponse,
} from 'stytch/types/lib/magic_links'

const sampleResp: MagicLinksAuthenticateResponse = {
  status_code: 200,
  request_id: 'mock-request-id',
  user_id: 'mock-user-id',
  method_id: 'mock-method-id',
  session_token: 'mock-token',
}

console.log(sampleResp);
```

<img width="763" alt="image" src="https://user-images.githubusercontent.com/89937743/166083286-564440ed-eb41-4c56-97e6-0dbdf53c63b7.png">

Do you have any examples of client libraries that do a really good job of this? We'd love to take a look for inspiration and see where we can improve.

**Comment by anthonyshew** [comment]
Oops, sorry, @max-stytch, forgot I had this out for you guys. I actually just ran into it in my codebase again and so I remembered to come back here!

Ah, that's actually exactly what I was looking for! I believe I would have benefitted from documentation around this, then, so that I would have known where to look. The code is apparently there; I just didn't know where to find it!

But at least I can fix it now for this codebase. 😁

**Comment by max-stytch** [comment]
Hey @hahnbeelee  - the B2BClient should be a top-level export on the `stytch` package - so you should be able to use it without importing the deeply nested packages.
```
import * as stytch from 'stytch';

let client: stytch.B2BClient;
```

You can also use `Awaited<ReturnType<typeof func>>` to capture return type values - we have an example [here](https://github.com/stytchauth/stytch-b2b-nextjs-example/blob/main/lib/loadStytch.ts#L1-L18).

Does that work for you?

🤞 we will be redoing our type exports with our next major version launch, which coincides with the deprecation of Node 14 in April

**Comment by max-stytch** [comment]
@anthonyshew All our types are publicly available with our **7.0.0** release.
```ts
import type { B2CMagicLinksAuthenticateResponse } from 'stytch'
```
